src/approaches/grammar_search_invention_approach.py

```diff
@@ -656,12 +657,20 @@ def _evaluate_with_operators(self, predicates: FrozenSet[Predicate],
         # Lower scores are better.
         scores = {name: 0.0 for name in self.heuristic_names}
         seen_demos = 0
+        seen_nondemos = 0
+        max_demos = CFG.grammar_search_heuristic_based_max_demos
+        max_nondemos = CFG.grammar_search_heuristic_based_max_nondemos
         for traj, atoms_sequence in pruned_atom_data:
-            if not traj.is_demo:  # we only care about demonstrations
+            # Skip this trajectory if it's not a demo and we only want demos,
+            # or if we've exceeded the budget for demos or nondemos.
+            if (self.demos_only and not traj.is_demo) or (
```

Comment:
confusing; let's write as 3 separate `if ...: continue`

Reply:
I think 2 is clear, so I'm going to do that